lib/Core/PTreeWriter.cpp

```diff
@@ -0,0 +1,169 @@
+//===-- PTreeWriter.cpp ---------------------------------------------------===//
+//
+//                     The KLEE Symbolic Virtual Machine
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "PTreeWriter.h"
+
+#include "PTree.h"
+#include "klee/Support/ErrorHandling.h"
+
+using namespace klee;
+
+
+void prepare_statement(sqlite3 * db, std::string &query, sqlite3_stmt ** stmt) {
+#ifdef SQLITE_PREPARE_PERSISTENT
+  if (sqlite3_prepare_v3(db, query.c_str(), -1, SQLITE_PREPARE_PERSISTENT, stmt, nullptr) != SQLITE_OK) {
+#else
+  if (sqlite3_prepare_v3(db, query.c_str(), -1, 0, stmt, nullptr) != SQLITE_OK) {
+#endif
+    klee_warning("Process tree database: can't prepare query: %s [%s]", sqlite3_errmsg(db), query.c_str());
+    sqlite3_close(db);
+    klee_error("Process tree database: can't prepare query: %s", query.c_str());
+  }
+}
+
+
+PTreeWriter::PTreeWriter(const std::string &dbPath) {
+  // create database file
+  if(sqlite3_open(dbPath.c_str(), &db) != SQLITE_OK)
+    klee_error("Can't create process tree database: %s", sqlite3_errmsg(db));
+
+  // - set options: asynchronous + WAL
+  char * errMsg = nullptr;
+  if (sqlite3_exec(db, "PRAGMA synchronous = OFF;", nullptr, nullptr, &errMsg) != SQLITE_OK) {
+    klee_warning("Process tree database: can't set option: %s", errMsg);
+    sqlite3_free(errMsg);
+  }
+  if (sqlite3_exec(db, "PRAGMA journal_mode = WAL;", nullptr, nullptr, &errMsg) != SQLITE_OK) {
+    klee_warning("Process tree database: can't set option: %s", errMsg);
+    sqlite3_free(errMsg);
+  }
+
+  // - create table
+  std::string query = "CREATE TABLE IF NOT EXISTS nodes ("
+                      "ID INT PRIMARY KEY, stateID INT, leftID INT, rightID INT,"
+                      "asmLine INT, branchType INT, terminationTypeMask INT);";
+  char * zErr = nullptr;
+  if (sqlite3_exec(db, query.c_str(), nullptr, nullptr, &zErr) != SQLITE_OK) {
+    klee_warning("Process tree database: initialisation error: %s", zErr);
+    sqlite3_free(zErr);
+    sqlite3_close(db);
+    klee_error("Process tree database: initialisation error.");
+  }
+
+  // create prepared statements
+  // - insertStmt
```

Comment:
These comments are excesive. Also why do you have this `query` variable, I would find it much nicer if you would put the string directly into the function call.